### ISSUE TITLE
Compiler errors for valid query expressions

### ISSUE THREAD

**Issue by rdhananjaya** [ballerina-platform/ballerina-lang#33707]
**Description:**
```ballerina
type BddPath record {|
    int[] pos = [1];
    int[] neg = [1];
|};

public type MappingAlternative record {|
    int semType;
    int[] pos;
    int[] neg;
|};

public function main() {
    BddPath[] paths = [];

    MappingAlternative[] alts = [];
    foreach var {pos, neg} in paths {
        alts.push({
            semType: 1,
            pos: (from var atom in pos select atom), // get compiler errors when parenthesis is removed
            neg: (from var atom in neg select atom)
        });
    }
}
```

**Steps to reproduce:**

**Affected Versions:**

**OS, DB, other environment details and versions:**

**Related Issues (optional):**

**Suggested Labels (optional):**

**Suggested Assignees (optional):**

**Comment by MaryamZi** [comment]
Seems to have been fixed in 2201.0.0. Keeping the issue open to add tests.

**Comment by KavinduZoysa** [comment]
Fixed in Ballerina 2201.3.0. Therefore closing the issue.
